### PR TITLE
Add support for exporting GLU to ONNX

### PR DESCRIPTION
**Pull request by bddppq**
@ezyang

### CODE REVIEW

File: torch/nn/_functions/thnn/auto_symbolic.py [hunk 1]
```diff
@@ -12,7 +12,17 @@ def leakyrelu_symbolic(g, input, negative_slope, inplace=False):
     return g.op("LeakyRelu", input, alpha_f=negative_slope)
 
 
+def glu_symbolic(g, input, dim):
+    assert input.type().sizes()[dim] % 2 == 0
```
Comment (ezyang) [inline torch/nn/_functions/thnn/auto_symbolic.py]: Is this a "real assert", or a "we don't support it if this condition doesn't hold" assert? If the latter, replace it with something that throws an exception.

**Comment by ezyang** [comment]
LGTM.

File: torch/nn/_functions/thnn/auto_symbolic.py (hunk 1 above)
Comment (bddppq) [reply torch/nn/_functions/thnn/auto_symbolic.py]: It's a real assert https://github.com/colesbury/pytorch/blob/6011d52fc12036b4b244a09e1c9cb226612e640a/torch/lib/THNN/generic/GatedLinearUnit.c#L14

File: torch/nn/_functions/thnn/auto_symbolic.py [hunk 2]
```diff
@@ -12,7 +12,17 @@ def leakyrelu_symbolic(g, input, negative_slope, inplace=False):
...
+
+    split_op = g.op('Split', input, axis_i=dim)
+    first = g.createSelect(split_op, 0)
+    second = g.createSelect(split_op, 1)
```
Comment (apaszke) [inline torch/nn/_functions/thnn/auto_symbolic.py]: I think this could have been just `first, second = g.op('Split', input, axis_i=dim, outputs=2)`

Comment (bddppq) [reply torch/nn/_functions/thnn/auto_symbolic.py]: Aha definitely better. I was looking for a way to specify the number of outputs.

**Pull request merged**